docs/application-developers-reference-guide.md

```diff
@@ -0,0 +1,83 @@
+
+To automate the Let's Encrypt certificate provisioning you should add the annotation `kubernetes.io/tls-acme: "true"` to the ingress resource of your application. Optionally you could add the annotation `kubernetes.io/ingress.class: "nginx"` to designate that the ingress resource should only be handled by the NGINX controller.
+
+For example, with the following snippet for a k8s ingress resource, BKPR will automatically update the DNS records for `myapp.mydomain.com` and request Let's Encrypt for a valid TLS certificate for your application, following which you would be able to access the application securely over the Internet.
+
+```yaml
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: "myapp-ingress"
+  labels:
+    name: "myapp"
+  annotations:
+    kubernetes.io/ingress.class: "nginx"
+    kubernetes.io/tls-acme: "true"
+spec:
+  tls:
+  - hosts:
+    - myapp.mydomain.com
+    secretName: myapp-tls
+  rules:
+  - host: myapp.mydomain.com
+    http:
+      paths:
+        - path: /
+          backend:
+            serviceName: myapp-svc
+            servicePort: 80
+```
+
+## Logging
+
+BKPR installs [Elasticsearch](https://elastic.co/products/elasticsearch) in the cluster which provides a RESTful search and analytics engine for processing your application logs. The stack uses [Fluentd](https://www.fluentd.org/) to capture the standard output of every container running in the cluster and persistent storage to store historical data of the cluster container logs.
+
+For Fluentd to be able to capture your applications logs, as a cloud-native application developer, all you need to do is configure your application to output the logs to the standard output of the application container.
+
+[Kibana](https://www.elastic.co/products/kibana) installed in the cluster lets you visualize your Elasticsearch data and navigate the Elastic Stack. You can use it to debug your application when any abnormal application behavior is noticed.
+
+## Monitoring
+
+Instrumentation and monitoring in BKPR is made possible by [Prometheus](https://prometheus.io/) which is a popular open-source monitoring system and time series database written in Go.
+
+By default, BKPR collects a lot of instrumentation data about your cluster. As a cloud-native application developer, you should instrument your own application code to take advantage of the built in monitoring so that you are alerted when abnormal application behavior is detected.
+
+If your application exports instrumentation data, specify the `prometheus.io/scrape: "true"` annotation so that the instrumentation data is scraped by Prometheus. Additionally you can specify the annotations `prometheus.io/port` to specify the port for the metrics endpoint and `prometheus.io/path` to specify the path of the metrics endpoint.
```

Comment:
As I see it, instrumentation data has to be exported in a particular way / format. Right? If so, it would be nice to provide pointers to Prometheus exporters clients libraries in languages like C++, Python, Java and Go.